tbxforms/fields.py

```diff
@@ -52,21 +71,47 @@ def __init__(self, **kwargs):
                 label=_("Day"),
                 error_messages={"incomplete": _("Enter the day of the month")},
                 validators=[
-                    RegexValidator(r"^[0-9]+$", _("Enter a valid date"))
+                    RegexValidator(r"^[0-9]+$", _("Enter a valid date")),
+                    StringMinValueValidator(
+                        1, _("Enter a value greater than 1")
+                    ),
+                    StringMaxValueValidator(
+                        31, _("Enter a value less than 31")
+                    ),
                 ],
             ),
             forms.CharField(
                 label=_("Month"),
                 error_messages={"incomplete": _("Enter the month")},
                 validators=[
-                    RegexValidator(r"^[0-9]+$", _("Enter a valid month"))
+                    RegexValidator(r"^[0-9]+$", _("Enter a valid month")),
+                    StringMinValueValidator(
+                        1, _("Enter a value greater than 1")
+                    ),
+                    StringMaxValueValidator(
+                        12, _("Enter a value less than 12")
+                    ),
```

Comment:
**Suggestion**: Can we reword this to avoid the literal implication that 1 and 12 are not permitted, but still use the numbers 1 and 12 in the message?

E.g. "Enter a value not more than 12" or "Enter a value up to 12"?

- [Error Messages | GOV.UK Design System](https://design-system.service.gov.uk/components/error-message/#use-instructions-and-descriptions) suggests "Month must be 12 or less"
- [Date Input | GOV.UK Design System](https://design-system.service.gov.uk/components/date-input/#if-the-date-entered-cannot-be-correct) suggests "(field name) must be a real date", while highlighting just that sub field if it's just one field that's wrong, and highlighting the whole compound field if it's unclear or more than one is wrong

I reckon "Month must be 12 or less" etc. works well here.

NB this comment applies to Day and Year too.

Reply:
This is resolved in db65d80.